database/ffldb/dbcache.go

```diff
@@ -483,12 +483,15 @@ func (c *dbCache) commitTreaps(pendingKeys, pendingRemove TreapForEacher) error
 	})
 }
 
-// flush flushes the database cache to persistent storage.  This involes syncing
+// flush flushes the database cache to persistent storage.  This involves syncing
 // the block store and replaying all transactions that have been applied to the
 // cache to the underlying database.
 //
 // This function MUST be called with the database write lock held.
 func (c *dbCache) flush() error {
+	c.cacheLock.Lock()
+	defer c.cacheLock.Unlock()
```

Comment:
Haven't checked the details, my main question is whether this will create a deadlock?